tofu/geom/_core.py

```diff
@@ -5760,6 +5760,86 @@ def calc_min_rho_from_Plasma2D(self, plasma, t=None, log='min',
                                     for vv in np.split(val, lind, axis=-1)])
         return vals, pts, t
 
+    def get_inspector(self, ff):
+        out = inspect.signature(ff)
+        pars = out.parameters.values()
+        na = np.sum([(pp.kind == pp.POSITIONAL_OR_KEYWORD
+                      and pp.default is pp.empty) for pp in pars])
+        kw = [pp.name for pp in pars if (pp.kind == pp.POSITIONAL_OR_KEYWORD
+                                         and pp.default is not pp.empty)]
+        return na, kw
+
+    def check_ff(self, ff, t=None, ani=None):
+        time_steps = -1
+        # .. Checking basic definition of function ..........................
+        str_error = "Input emissivity function (ff): "
+        assert hasattr(ff, '__call__'), (str_error
+                                         + " must be a callable (function)!")
+        npos_args, kw = self.get_inspector(ff)
+        assert npos_args == 1, (str_error
+                              + " must take only one positional argument:"
+                              + " ff(Pts)!")
+        assert 't' in kw, (str_error
+                           + " must have kwarg 't=None' for time vector!")
+        is_t_type_valid = (type(t) in [int, float, np.int64, np.float64]
+                           or hasattr(t, '__iter__'))
+        assert t is None or is_t_type_valid, (str_error
+                                              + "Arg t must be None,"
+                                              + " a scalar or an iterable!")
+        # .. Testing outputs ...............................................
+        test_pts = np.array([[1, 2], [3, 4], [5, 6]])
+        npts = test_pts.shape[1]
+        try:
+            out = ff(test_pts, t=t)
+        except Exception:
+            assert False, (str_error
```

Comment:
To force raise an exception just use
`raise Exception(msg)`

(assert False if overkill)